x/curating/types/key.go

```diff
@@ -0,0 +1,62 @@
+package types
+
+import (
+	"encoding/binary"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// ModuleName is the name of the module
+	ModuleName = "curating"
+
+	// StoreKey to be used when creating the KVStore
+	StoreKey = ModuleName
+
+	// RouterKey to be used for routing msgs
+	RouterKey = ModuleName
+
+	// QuerierRoute to be used for querierer msgs
+	QuerierRoute = ModuleName
+
+	// RewardPoolName is the name for the module account for the reward pool
+	RewardPoolName = "reward_matching_pool"
+
+	// DefaultStakeDenom is the staking denom for the zone
+	DefaultStakeDenom = "ufuel"
+)
+
+var (
+	// 0x00 | vendor_id | post_id
+	KeyPrefixPost = []byte{0x00}
+
+	// 0x01 | vendor_id | post_id | curator
+	KeyPrefixUpvote = []byte{0x01}
+
+	// 0x02 | format(curation_end_time) -> []VPPair
+	KeyPrefixCurationQueue = []byte{0x02}
+)
+
+var lenTime = len(sdk.FormatTimeBytes(time.Now()))
+
+func PostKey(vendorID uint32, postID string) []byte {
+	vendorIDBz := uint32ToBigEndian(vendorID)
+	return append(KeyPrefixPost, append(vendorIDBz, []byte(postID)...)...)
+}
+
+func UpvoteKey(vendorID uint32, postID string, curator sdk.AccAddress) []byte {
```

Comment:
`postID` being a string with a variable length here is a bit precarious. There's the potential for false matches when iterating if another longer `postID` happened to partially match. Maybe it should be hashed. Thoughts? Alternatives?

Reply:
yeah I was thinking the same , hashing could work (adding cpu time )   downside is that if we have iterators that go over all posts these may result in a different order, but I dont think we need to iterate over posts and only by a specific`post_id/<votes>`

Reply:
Will use MD5 as it's fast and produces 128-bit hashes. Since it's always paired with a vendor_id, chances of collisions are low.